Src/FluentAssertions/Primitives/TimeOnlyAssertions.cs

```diff
@@ -127,6 +127,94 @@ public AndConstraint<TAssertions> Be(TimeOnly? expected, string because = "", pa
         return new AndConstraint<TAssertions>((TAssertions)this);
     }
 
+    /// <summary>
+    /// Asserts that the current <see cref="TimeOnly"/>  is within the specified time
+    /// from the specified <paramref name="nearbyTime"/> value.
+    /// </summary>
+    /// <param name="nearbyTime">
+    /// The expected time to compare the actual value with.
+    /// </param>
+    /// <param name="precision">
+    /// The maximum amount of time which the two values may differ.
+    /// </param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndConstraint<TAssertions> BeCloseTo(TimeOnly nearbyTime, TimeSpan precision, string because = "",
+        params object[] becauseArgs)
+    {
+        if (precision < TimeSpan.Zero)
+        {
+            throw new ArgumentOutOfRangeException(nameof(precision), $"The value of {nameof(precision)} must be non-negative.");
+        }
+
+        TimeOnly minimumValue = nearbyTime.Add(-precision);
+        TimeOnly maximumValue = nearbyTime.Add(TimeSpan.FromTicks(precision.Ticks + 1));
+
+        TimeSpan? difference = (Subject != null)
+            ? new[] { Subject.Value - nearbyTime, nearbyTime - Subject.Value }.Min()
```

Comment:
I am not sure about this line..

Was by far the least code :)

but it works like I would expect it to work.

Reply:
The alternative would be:
`TimeSpan.FromTicks(Math.Min((Subject.Value - nearbyTime).Duration().Ticks, (nearbyTime - Subject.Value).Duration().Ticks))`

Reply:
I think this works

```cs
TimeSpan? difference = Subject.HasValue ? MinimumDifference(Subject.Value, nearbyTime) : null;
...
private static TimeSpan? MinimumDifference(TimeOnly a, TimeOnly b)
{
    var diff1 = a - b;
    var diff2 = b - a;

    return diff1 < diff2 ? diff1 : diff2;
}
```